engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformulamulticolumn/RollingFormulaMultiColumnOperator.java

```diff
@@ -0,0 +1,394 @@
+        }
+
+        @Override
+        public void writeToOutputColumn(@NotNull final RowSequence inputKeys) {
+            primitiveOutputSource.fillFromChunk(outputFillContext, outputValues, inputKeys);
+        }
+
+        @Override
+        public void reset() {
+            // Clear all the ring buffers for re-use
+            for (RingBufferWindowConsumer consumer : inputConsumers) {
+                consumer.reset();
+            }
+            nullCount = 0;
+        }
+
+        @Override
+        public void close() {
+            outputValues.close();
+            outputFillContext.close();
+        }
+    }
+
+    public RollingFormulaMultiColumnOperator(
+            @NotNull final MatchPair pair,
+            @NotNull final String[] affectingColumns,
+            @Nullable final String timestampColumnName,
+            @NotNull final String formula,
+            final long reverseWindowScaleUnits,
+            final long forwardWindowScaleUnits,
+            @NotNull final TableDefinition tableDef,
+            @NotNull final QueryCompilerRequestProcessor compilationProcessor) {
+        super(pair, affectingColumns, timestampColumnName, reverseWindowScaleUnits, forwardWindowScaleUnits, true);
+        this.tableDef = tableDef;
+
+        final String outputColumnName = pair.leftColumn;
+
+        formulaColumn = FormulaColumn.createFormulaColumn(outputColumnName, formula);
+
+        final Map<String, ColumnDefinition<?>> columnDefinitionMap = tableDef.getColumnNameMap();
+
+        // Create a column definition map composed of vector types for the formula.
+        // TODO: do this ahead of time and provide it as an alternate table definition?
+        final Map<String, ColumnDefinition<?>> vectorColumnNameMap = new HashMap<>();
+        columnDefinitionMap.entrySet().forEach(e -> {
+            final ColumnDefinition<?> columnDef = ColumnDefinition.fromGenericType(
+                    e.getKey(), VectorFactory.forElementType(e.getValue().getDataType()).vectorType());
+            vectorColumnNameMap.put(e.getKey(), columnDef);
+        });
+
+        // Get the input column names and data types from the formula.
+        inputColumnNames = formulaColumn.initDef(vectorColumnNameMap, compilationProcessor).toArray(String[]::new);
+        inputColumnTypes = new Class[inputColumnNames.length];
+        inputComponentTypes = new Class[inputColumnNames.length];
+        inputVectorTypes = new Class[inputColumnNames.length];
+
+        for (int i = 0; i < inputColumnNames.length; i++) {
+            final ColumnDefinition<?> columnDef = columnDefinitionMap.get(inputColumnNames[i]);
+            inputColumnTypes[i] = columnDef.getDataType();
+            inputComponentTypes[i] = columnDef.getComponentType();
+            inputVectorTypes[i] = vectorColumnNameMap.get(inputColumnNames[i]).getDataType();
+        }
+    }
+
+    @Override
+    public UpdateByOperator copy() {
+        return null;
```

Comment:
This feels weird, should it be an IllegalStateException

Reply:
No, this is an oversight.  It should return a legitimate copy of this operator.